samcli/commands/validate/validate.py

```diff
@@ -170,20 +199,97 @@ def _lint(ctx: Context, template: str, template_path: str) -> None:
         cfn_lint_logger.propagate = True
         cfn_lint_logger.setLevel(logging.DEBUG)
 
+    print(f"Debug info: initial linter_config = {linter_config}")
+
+    # Initialize variable to handle both options together
+    rules_to_append = []
+    
+    # Support for previous serverless_rules option (deprecated)
+    if serverless_rules:
```

Comment:
Just to emphasize what I mentioned before, don't add any of the `serverless_rules` code.